python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -681,6 +684,10 @@ def __init__(self, **data: Any):
             nested_resources={k: v for k, v in resource_pointers.items()},
             resource_context=None,
         )
+        self.__post_init__()
```

Comment:
One immediate question is where to invoke this - right now it's after every `__init__` call, but this means it's called both at instantiation time (e.g. in Definitons) and at run time when the resource is fully setup (e.g. env vars populated, nested resources populated).

We may want to only call it as part of the resource function, e.g. after the resource is set up at run time